File: seeker/snippet/clock.py
#date: 2021-12-10T17:03:12Z
#url: https://api.github.com/gists/fcff4e15b84e3393ceefe578d6f1ecfe
#owner: https://api.github.com/users/Este30

from math import *
from kandinsky import *

option_logo = [
[0,0,0,0,0,0,1,1,1,0,0,0,0,0,0],
[0,0,0,0,1,1,0,0,0,1,1,0,0,0,0],
[0,0,1,1,0,0,0,0,0,0,0,1,1,0,0],
[0,0,1,0,0,0,0,1,0,0,0,0,1,0,0],
[0,1,0,0,0,0,0,1,0,0,0,0,0,1,0],
[0,1,0,0,0,0,0,1,0,0,0,0,0,1,0],
[1,0,0,0,0,0,0,1,0,0,0,0,0,0,1],
[1,0,0,0,0,0,0,1,1,1,1,1,0,0,1],
[1,0,0,0,0,0,0,0,0,0,0,0,0,0,1],
[0,1,0,0,0,0,0,0,0,0,0,0,0,1,0],
[0,1,0,0,0,0,0,0,0,0,0,0,0,1,0],
[0,0,1,0,0,0,0,0,0,0,0,0,1,0,0],
[0,0,1,1,0,0,0,0,0,0,0,1,1,0,0],
[0,0,0,0,1,1,0,0,0,1,1,0,0,0,0],
[0,0,0,0,0,0,1,1,1,0,0,0,0,0,0],
]

def image_display(image_list, x, y, pos_x, pos_y):
    for x in range (0, x):
        for y in range (0, y):
            

File: seeker/snippet/clock.go
//date: 2024-03-27T17:04:12Z
//url: https://api.github.com/gists/3f8b8731054006d8fbd7b13b06d1b632
//owner: https://api.github.com/users/Farhaduneci

package main

import (
	"fmt"
)

type Clock struct {
	hours   int
	minutes int
}

func New(hours, minutes int) Clock {
	h := (hours + minutes/60) % 24
	m := minutes % 60

	if m < 0 {
		m += 60
		h -= 1
	}
	if h < 0 {
		h += 24
	}

	return Clock{h, m}
}

func (c Clock) Add(minutes int) Clock {
	return New(c.hours, c.minutes+minutes)
}

func (c Clock) Subtract(minutes int) Clock {
	return New(c.hours, c.minutes-minutes)
}

func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hours, c.minutes)
}

